src/ImageSharp/Image/ImageBase{TColor}.cs

```diff
@@ -162,13 +162,15 @@ public virtual PixelAccessor<TColor> Lock()
         internal void SwapPixelsBuffers(PixelAccessor<TColor> pixelSource)
         {
             Guard.NotNull(pixelSource, nameof(pixelSource));
-            Guard.IsTrue(pixelSource.PooledMemory, nameof(pixelSource.PooledMemory), "pixelSource must be using pooled memory");
 
+            // TODO: Was this check really useful? If yes, we can define a bool PixelAccessor<TColor>.IsBoundToImage to re-introduce it:
```

Comment:
If it's unused then yes. I introduced it ages ago and the functionality has since been superseded with better objects.